cmd/snap/cmd_snap_op.go

```diff
@@ -350,13 +350,27 @@ func maybeWithSudoSecurePath() bool {
 	return release.DistroLike("fedora", "opensuse", "debian")
 }
 
+func hasWaitTasks(chg *client.Change) bool {
+	for _, t := range chg.Tasks {
+		if t.Status == "Wait" {
```

Comment:
checking just for this has a bit the problem that other independent part of the change might still be able to progress, which might make rebooting a bit premature

Reply:
I made the code more precise but I do feel a bit uneasy about this, given that this (new code) is not very nice and also the progress in the wait() mixin is not very nice maybe this indicates this approach is a dead-end and we should do the more complicated approach you suggested in https://github.com/snapcore/snapd/pull/12417#discussion_r1067201831

Reply:
they way Change.Status is defined makes the most sense if all the task are interdependent, but this is not the case anymore, and Wait surfaces this